meson.build

```diff
@@ -18,6 +18,7 @@ add_project_arguments(cc.get_supported_arguments([
 	'-Wno-missing-field-initializers',
 	'-Wno-unused-parameter',
 	'-D_POSIX_C_SOURCE=200809L',
+	'-D_GNU_SOURCE',
```

Comment:
Fails on FreeBSD as `usleep` [was removed](https://pubs.opengroup.org/onlinepubs/9699919799/xrat/V4_xsh_chap01.html#tag_22_01_01_05) in POSIX 2008 (aka X/Open Issue 7). Maybe downgrade `-D_POSIX_C_SOURCE=200809L` to `-D_POSIX_C_SOURCE=200112L` instead of cheating on Linux via `-D_GNU_SOURCE`.
```c
$ meson setup _build && meson compile -C _build
../src/screencast/fps_limit.c:29:4: error: implicit declaration of function 'usleep' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                        usleep(delay_us);
                        ^
../src/screencast/fps_limit.c:29:4: note: did you mean 'sleep'?
/usr/include/unistd.h:365:15: note: 'sleep' declared here
unsigned int     sleep(unsigned int);
                 ^
1 error generated.
```

Reply:
The suggested flag makes the compilation fail for me on Arch Linux with the exact above error message. I don't have access to FreeBSD, but I could spin one up in a VM if we cannot resolve this otherwise.

I must admit that I have no idea how to fix this. I was taking the above flag from wayvnc's build: https://github.com/any1/wayvnc/blob/13323a742f1449355852b2501b25d8df804cefee/meson.build#L17

Can anybody please help?

Reply:
> 4.3BSD, POSIX.1-2001.  POSIX.1-2001 declares this function obsolete; use nanosleep(2) instead.  POSIX.1-2008 removes the specification of usleep()

Reply:
@emersion Thank you! I just pushed another version that uses `nanosleep`, and it actually works without changes to `build.meson`. I hope it works on FreeBSD also.

Reply:
I confirm, builds fine on FreeBSD. Rebase after cf3e02695e3d if you want to see build log.

Reply:
Glad to hear that! I am marking this comment as resolved, then.

I also just fixup'd the fix on the build and `nanosleep` into the first commit, and rebased the whole branch onto the latest master.